### PR TITLE
shuffle restore works

### PR DESCRIPTION
**Pull request by hassantahir**
This PR implements the ability to restore a deck to its original order after it has been shuffled. Previously, there was no way to restore the deck to its default order after shuffling.

@MichaelDHilborn-Harvard @jazahn Can you review this when you get a chance?

---
[FLASH-247](https://jira.huit.harvard.edu/browse/FLASH-247)

### CODE REVIEW

**Comment by arthurian** [comment]
@hassantahir Just to clarify, did you need to push up one more batch of commits, or is this all set?

**Comment by hassantahir** [comment]
It's all set!
ᐧ

**Comment by arthurian** [comment]
Excellent! @MichaelDHilborn-Harvard  and @jazahn, Hassan has confirmed this is all set to go.  Can you review when you get a chance?

File: harvardcards/static/js/components/slider/DeckSlider.js [hunk 1]
```diff
@@ -258,10 +262,10 @@ define(['jquery', 'microevent', 'components/slider/Slider'], function($, MicroEv
 	DeckSlider.prototype.shuffle = function() {
 		var cards  = $(this.el).find("ul#cards");
 		var child = cards.children();
-		var random_index;
+		var random_index = -1;
 
 		while (child.length) {
-			random_index = Math.floor(Math.random() *  child.length);
+            random_index = Math.floor(Math.random() *  child.length);
```
Comment (MichaelDHilborn-Harvard) [inline harvardcards/static/js/components/slider/DeckSlider.js]: Looks like the indents were inappropriately removed here?

**Comment by MichaelDHilborn-Harvard** [comment]
Overall, looks good, though it looks like some of the tabs/indentation are not aligned correctly. Can they be fixed or maybe they are just artifacts? Sorry to be picky.

**Comment by arthurian** [comment]
@hassantahir Can you fix the indents in this PR when you get a chance (see Mike's comment)?

**Comment by hassantahir** [comment]
I just checked the javascript file and the indents are correctly aligned. Not sure why it isnt showing up that way here.

**Comment by arthurian** [comment]
@hassantahir Fixed it. Looks like the javascript files are indented with tabs, and your editor is indenting with spaces. I just converted the spaces to tabs in a few places. It would probably be better if the whole project was one or the other, but as it is, the js files are tabs and the python files are spaces.

**Pull request merged**